build.sbt

```diff
@@ -1,92 +1,81 @@
-  pomExtra :=
-    <url>https://github.com/sparsetech/pine</url>
-    <licenses>
-      <license>
-        <name>Apache-2.0</name>
-        <url>https://www.apache.org/licenses/LICENSE-2.0.html</url>
-      </license>
-    </licenses>
-    <scm>
-      <url>git@github.com:sparsetech/pine.git</url>
-    </scm>
-    <developers>
-      <developer>
-        <id>tindzk</id>
-        <name>Tim Nieradzik</name>
-        <url>http://github.com/tindzk</url>
-      </developer>
-    </developers>
-)
+  // See https://github.com/sbt/sbt/pull/2659
+  incOptions := incOptions.value.withLogRecompileOnMacro(false)
+}
 
-lazy val root = project.in(file("."))
-  .aggregate(pineJS, pineJVM, pineNative)
-  .settings(SharedSettings: _*)
-  .settings(skip in publish := true)
+val Pre13Settings = nocomma {
+  scalaOrganization := "org.typelevel"
+  scalacOptions += "-Yliteral-types"
 
-val convertMDN = taskKey[Unit]("Generate MDN bindings")
+  libraryDependencies ++= Seq(
+    scalaOrganization.value % "scala-reflect" % scalaVersion.value % "provided",
+    scalaOrganization.value % "scala-compiler" % scalaVersion.value % "provided",
+    compilerPlugin(("org.scalamacros" %% "paradise" % V.paradise).cross(CrossVersion.patch))
+  )
+}
 
-lazy val pine = crossProject(JSPlatform, JVMPlatform, NativePlatform)
-  .in(file("."))
-  .settings(SharedSettings: _*)
-  .settings(
-    addCompilerPlugin("org.scalamacros" % "paradise" % Paradise cross CrossVersion.patch),
-    libraryDependencies ++= Seq(
-      scalaOrganization.value % "scala-reflect" % scalaVersion.value % "provided",
-      scalaOrganization.value % "scala-compiler" % scalaVersion.value % "provided"
-    ),
-    convertMDN := MDNParser.createFiles(new File("shared/src/main/scala"))
+val JvmSettings = nocomma {
+  libraryDependencies ++= Seq(
+    "org.scalatest"  %%% "scalatest"   % V.scalaTest  % "test",
+    "org.scalacheck" %%% "scalacheck"  % V.scalaCheck % "test",
   )
-  .jvmSettings(
-    libraryDependencies ++= Seq(
-      "org.scalatest"  %% "scalatest"  % ScalaTest  % "test",
-      "org.scalacheck" %% "scalacheck" % ScalaCheck % "test"
-    )
-  ).jsSettings(
-    // From https://github.com/scala-js/scala-js/pull/2954
-    libraryDependencies := libraryDependencies.value.filterNot(_.name == "scalajs-compiler"),
-    addCompilerPlugin("org.scala-js" % "scalajs-compiler" % scalaJSVersion cross CrossVersion.patch),
+}
 
-    libraryDependencies ++= Seq(
-      "org.scala-js"   %%% "scalajs-dom" % ScalaJsDom,
-      "org.scalatest"  %%% "scalatest"   % ScalaTest  % "test",
-      "org.scalacheck" %%% "scalacheck"  % ScalaCheck % "test"
-    ),
-    jsEnv in Test := new org.scalajs.jsenv.jsdomnodejs.JSDOMNodeJSEnv,
-    scalaJSStage in Global := FastOptStage
-  ).nativeSettings(
-    excludeFilter in Test := "*",
+val JsSettings = JvmSettings ++ nocomma {
+  // From https://github.com/scala-js/scala-js/pull/2954
+  libraryDependencies := libraryDependencies.value.filterNot(_.name == "scalajs-compiler")
```

Comment:
This line should only be needed for < 2.13.

Reply:
Yes, I was considering that, planned on revisiting alongside travis